pkg/synopsysctl/cmd_destroy.go

```diff
@@ -323,8 +216,100 @@ func deleteCrd(crd string, namespace string) {
 	}
 }
 
-func init() {
-	rootCmd.AddCommand(destroyCmd)
+func deleteClusterRoleBinding(namespace string) {
+	// delete cluster role bindings
+	clusterRoleBindings, _, err := util.GetOperatorRoleBindings(kubeClient, namespace)
+	if err != nil {
+		log.Errorf("error getting role binding or cluster role binding due to %+v", err)
+	}
+
+	for _, clusterRoleBinding := range clusterRoleBindings {
+		crb, err := util.GetClusterRoleBinding(kubeClient, clusterRoleBinding)
+		if err != nil {
+			log.Errorf("unable to get cluster role binding '%s' due to %+v", clusterRoleBinding, err)
+		}
+		// check whether any subject present for other namespace before deleting them
+		newSubjects := []rbacv1.Subject{}
+		for _, subject := range crb.Subjects {
+			isExist := util.IsSubjectExistForOtherNamespace(subject, namespace)
+			if isExist {
+				newSubjects = append(newSubjects, subject)
+			}
+		}
+		if len(newSubjects) > 0 {
+			crb.Subjects = newSubjects
+			// update the cluster role binding to remove the old cluster role binding subject
+			log.Infof("updating cluster role binding '%s'", clusterRoleBinding)
+			_, err = util.UpdateClusterRoleBinding(kubeClient, crb)
+			if err != nil {
+				log.Errorf("unable to update cluster role binding '%s' due to %+v", clusterRoleBinding, err)
+			}
+		} else {
+			log.Infof("deleting cluster role binding '%s'", clusterRoleBinding)
+			err := util.DeleteClusterRoleBinding(kubeClient, clusterRoleBinding)
+			if err != nil {
+				log.Errorf("unable to delete cluster role binding '%s' due to %+v", clusterRoleBinding, err)
+			}
+		}
+	}
+}
+
+func deleteClusterRole(namespace string) {
+	// delete cluster roles
+	clusterRoles, _, err := util.GetOperatorRoles(kubeClient, namespace)
+	if err != nil {
+		log.Errorf("error getting role or cluster role due to %+v", err)
+	}
+
+	crbs, err := util.ListClusterRoleBindings(kubeClient, "app in (synopsys-operator,opssight)")
 
-	destroyCmd.Flags().BoolVarP(&isForceDestroy, "force", "f", isForceDestroy, "Forcefully destroy the Synopsys Operator in your cluster")
+	for _, clusterRole := range clusterRoles {
+		isExist := false
+		// check whether the cluster role is referenced in any cluster role binding
+		for _, crb := range crbs.Items {
+			if util.IsClusterRoleRefExistForOtherNamespace(crb.RoleRef, clusterRole, namespace, crb.Subjects) {
+				isExist = true
+			}
+		}
+		if !isExist {
+			log.Infof("deleting cluster role '%s'", clusterRole)
+			err := util.DeleteClusterRole(kubeClient, clusterRole)
+			if err != nil {
+				log.Errorf("unable to delete cluster role '%s' due to %+v", clusterRole, err)
+			}
+		}
+	}
+}
+
+// isDeleteOperatorResources check whether there is any custom resources are existing
+func isDeleteOperatorResources(crdNamespace string, crdNames []string) (bool, error) {
```

Comment:
Can we rename this function to something like "CustomResourceInstancesExist"? I got a little confused